### PR TITLE
GG-30837 Make .sh and .bat scripts for index reader and wal reader

### PR DESCRIPTION
**Pull request by JKonSir**

### CODE REVIEW

File: bin/index-reader.bat [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+::     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+::
+:: Unless required by applicable law or agreed to in writing, software
+:: distributed under the License is distributed on an "AS IS" BASIS,
+:: WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+:: See the License for the specific language governing permissions and
+:: limitations under the License.
+::
+::
+:: Grid command line loader.
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:17]: This comment is copy-pasted from `ignite.bat`, it doesn't make sense in the context of index-reader.

How about `Utility for reading index files` or something else describing functionality of this script?

The same is applicable to captions of other three files.

File: bin/index-reader.bat [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+if %MAJOR_JAVA_VER% LSS 8 (
+    echo %0, ERROR:
+    echo The version of JAVA installed in %JAVA_HOME% is incorrect.
+    echo Please point JAVA_HOME variable to installation of JDK 1.8 or later.
+    echo You can also download latest JDK at http://java.com/download.
+    goto error_finish
+)
+
+:: Check IGNITE_HOME.
+:checkIgniteHome1
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:65]: Do we actually need IGNITE_HOME to be set for index reader?

File: bin/index-reader.bat [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+if %ERRORLEVEL% equ 0 (
+    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
+) else (
+    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
+)
+
+::
+:: Uncomment to enable experimental commands [--wal]
+::
+:: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -DIGNITE_ENABLE_EXPERIMENTAL_COMMAND=true
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:144]: I don't think we need experimental commands support in this script.

File: bin/index-reader.bat [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+::
+:: Uncomment the following GC settings if you see spikes in your throughput due to Garbage Collection.
+::
+:: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -XX:+UseG1GC
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:149]: Is it applicable to index reader? If not please remove this from the script.

File: bin/index-reader.bat [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+::
+:: Uncomment if you get StackOverflowError.
+:: On 64 bit systems this value can be larger, e.g. -Xss16m
+::
+:: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Xss4m
+
+::
+:: Uncomment to set preference to IPv4 stack.
+::
+:: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Djava.net.preferIPv4Stack=true
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:160]: As far as I understand index reader doesn't need any network communication so this option isn't needed here.

File: bin/index-reader.bat [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+::
+:: Set main class to start service (grid node by default).
+::
+
+if "%MAIN_CLASS%" == "" set MAIN_CLASS=org.apache.ignite.internal.commandline.indexreader.IgniteIndexReader
+
+::
+:: Remote debugging (JPDA).
+:: Uncomment and change if remote debugging is required.
+:: set CONTROL_JVM_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8787 %CONTROL_JVM_OPTS%
```
Comment (sergey-chugunov-1985) [inline bin/index-reader.bat:184]: Don't think that we need remote debugging for index reader utility.

File: bin/wal-reader.bat [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    --illegal-access=permit ^
+    %CONTROL_JVM_OPTS%
+)
+
+if defined JVM_OPTS (
+    echo JVM_OPTS environment variable is set, but will not be used. To pass JVM options use CONTROL_JVM_OPTS
+    echo JVM_OPTS=%JVM_OPTS%
+)
+
+if "%INTERACTIVE%" == "1" (
```
Comment (sergey-chugunov-1985) [inline bin/wal-reader.bat:219]: I'm not sure if we need this check. Where INTERACTIVE mode is set and how two blocks of code in `then` and `else` branches differ from one another?

**Review by sergey-chugunov-1985 — CHANGES_REQUESTED** [review]
@konstantins304 , I left several comments in one file, they are also applicable to another ones too.

File: bin/index-reader.bat (hunk 1 above)
Comment (konstantins304) [reply bin/index-reader.bat]: fixed

File: bin/index-reader.bat (hunk 3 above)
Comment (konstantins304) [reply bin/index-reader.bat]: fixed

File: bin/index-reader.bat (hunk 4 above)
Comment (konstantins304) [reply bin/index-reader.bat]: fixed

File: bin/index-reader.bat (hunk 5 above)
Comment (konstantins304) [reply bin/index-reader.bat]: fixed

File: bin/index-reader.bat (hunk 6 above)
Comment (konstantins304) [reply bin/index-reader.bat]: fixed

File: bin/wal-reader.bat (hunk 7 above)
Comment (konstantins304) [reply bin/wal-reader.bat]: fixed

**Pull request merged**